core/trino-spi/src/main/java/io/trino/spi/exchange/Exchange.java

```diff
@@ -59,6 +59,12 @@
      */
     ExchangeSinkInstanceHandle instantiateSink(ExchangeSinkHandle sinkHandle, int taskAttemptId);
 
+    /**
+     * Refreshes sink instance handle. Called after we learn from the worker (via TaskStatus) that sink is permanently blocked.
+     * Method can be also proactively called by Coordinator without signal from the worker side.
```

Comment:
Actually it looks like we need to depend on the signal from worker side, as there is no flow on the coordinator side which would allow coordinator to learn that given `ExchangeSinkInstanceHandle` needs refreshing.
To make that work we would need to add some listener registration mechanism, so coordinator may register itself for information that we need a refresh. 
But I think it may no be necessary additional complexity and just waiting for information from worker should work fine. In rare cases it can add some extra latency as we learn that extra work needs to be done later, when we actually want to use the sink, but it should not be a big deal. WDT @arhimondr ?

Reply:
This can be delivered as part of `TaskStatus` that is refreshed every second. I think it should be fine.